src/common/resolve.js

```diff
@@ -4,9 +4,14 @@ let { resolve } = eval("require");
 
 // In the VS Code extension `require` is overridden and `require.resolve` doesn't support the 2nd argument.
 if (resolve.length === 1) {
-  const Module = eval("require")("module");
-  const createRequire = Module.createRequire || Module.createRequireFromPath;
-  resolve = createRequire(__dirname).resolve;
+  resolve = (id, options) => {
+    let baseDir;
+    if (options && options.paths && options.paths.length === 1) {
+      baseDir = options.paths[0];
+    }
+
+    return require("resolve").sync(id, { baseDir });
```

Comment:
With this conditional `require` call, we might face the problem you mentioned [here](https://github.com/prettier/prettier/pull/8036#issuecomment-614620428). We should check how it will look in the bundle, or better think of a way to test this fallback.

Reply:
Don't know, need check dist.

Reply:
I was trying to wrote a test couple days ago, but failed.

I tried to override `require.resolve` function before test, and restore after test.

Reply:
Overriding is too complicated. I suggest that we simply use an environment variable to trigger the fallback: `resolve.length === 1 || process.env.PRETTIER_FALLBACK_RESOLVE`.

Reply:
Yes, let's use env to run the test, but before that. I'm going to refactor production test first.